### DISCUSSION TITLE
[ZK Stack] - Withdrawal Finalizer not picking up withdrawals properly

### DISCUSSION THREAD

**Discussion by therealdannzor** [zkSync-Community-Hub/zksync-developers#457] [ZK Stack]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

1.78.0-nightly (d44e3b95c 2024-02-09)

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Setup:
* Already deployed zkSync hyperchain (our testnet)
* Local running instance of `dapp-portal`
* Running the binary of [zksync-withdrawal-finalizer](https://github.com/matter-labs/zksync-withdrawal-finalizer)

I have:
* deployed the `WithdrawalFinalizer` contract in `zksync-withdrawal-finalizer` successfully
* made sure that all the env variables are set up properly
* properly created and migrated to local Postgres database

Despite that, it doesn't perform automatic withdrawals on our zkSync hyperchain to L1 Sepolia.

### Expected Behavior

Claims are automatically performed on L1

### Current Behavior

<img width="652" alt="Screenshot 2024-04-10 at 11 25 32" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/35889839/47f44522-acc6-4285-a6a8-c8664655c853">

### Repository Link (if applicable)

https://github.com/matter-labs/zksync-withdrawal-finalizer

### Additional Details

* our hyperchain is not EIP-4844 compliant yet
* enabled `FINALIZE_ETH_TOKEN` as this only concerns ETH
* hardcoded [contract constants](https://github.com/matter-labs/zksync-withdrawal-finalizer/blob/main/contracts/WithdrawalFinalizer.sol#L12-L13) before deployment
* `WITHDRAWAL_FINALIZER_ACCOUNT_PRIVATE_KEY` is funded to make txs
* `DATABASE_URL` indexes events in my local database
* no attempts to send any txs from the backend service at all
* no observed calls to `CONTRACTS_WITHDRAWAL_FINALIZER_CONTRACT`
* `GAS_LIMIT=30000000` and `BATCH_FINALIZATION_GAS_LIMIT=10000000000000`
* tried `TX_RETRY_TIMEOUT_SECS=10..60`
* set correct L1 chain id: `wallet = wallet.clone().with_chain_id(11155111u64);`
* verified withdrawal finalizer contract on Sepolia etherscan
<br />

*Tables in my database*: <br />
<img width="195" alt="Screenshot 2024-04-10 at 20 31 01" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/35889839/6cf65c5a-0b46-40f4-b936-7c08ba25daac">  <br />

`l2_to_l1_events` and `tokens` are empty.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by bxpana** [comment]
Thanks for reporting, will check with the team on this and get back to you.

**Reply by therealdannzor** [reply]
Much appreciated.

Are you able to provide the env variables `GAS_LIMIT` and `BATCH_FINALIZATION_GAS_LIMIT` that your team is running in the deployed service? In your [docs](https://docs.zksync.io/build/support/faq.html#what-is-block-gas-limit-on-zksync-era) the amount `2^32` is mentioned but I've set it lower to avoid the error, encountered in my initial post (see edit). Based on experience, a lot of issues comes from gas configuration.

**Reply by therealdannzor** [reply]
To give an update, I don't think it has to do with the gas since the dry-run with `eth_call` works fine without reverting and I even added (superfluous) functionality by calling `estimate_gas()` to update the config dynamically. Despite that, all the withdrawals I observe are predicted to fail and the ones that I initiate myself are discovered by the `watcher` module but never acted on by the `finalizer` module.

**Reply by therealdannzor** [reply]
I also asked the dev for the project ([PR-498](https://github.com/matter-labs/zksync-withdrawal-finalizer/pull/398)) why the table `l2_to_l1_events` are no longer needed as it might be an explanation for why it doesn't work in our non-4844 compliant hyperchain.

**Reply by bxpana** [reply]
I'm still waiting to hear back from the team on this and will let you know.

**Reply by montekki** [reply]
Hi, @therealdannzor , thanks for reporting this. First we need to look at the transaction that initiate withdrawals and what logs they are generating. Can you provide such an example? Idk, if you have the block explorer for your chain deployed. But maybe you can just get transaction and logs for it from the RPC?

**Comment by therealdannzor** [comment]
x-post https://github.com/matter-labs/zksync-withdrawal-finalizer/issues/435
